### ISSUE TITLE
Suggested change produces different results than original

### ISSUE THREAD

**Issue by bkcsfi** [sourcery-ai/sourcery#70]
### Issue description or question

Please see screenshot below. The suggested change is what sourcery actually implements when the fix is applied.

Please note that the call to function `dot_getattr` is removed.

![image](https://user-images.githubusercontent.com/6412502/89435642-272f8e80-d713-11ea-92d1-bd2041956138.png)

existing code

```python
def assign_by_map(destination: Any, source: Any, attr_map: Dict[str, str]) -> Any:
    for dest_key in attr_map.keys():
        value = dot_getattr(source, attr_map[dest_key], None)
        if value:
            setattr(destination, dest_key, value)
    return destination
```

after applying quick fix

```python
def assign_by_map(destination: Any, source: Any, attr_map: Dict[str, str]) -> Any:
    for dest_key, value in attr_map.items():
        if value:
            setattr(destination, dest_key, value)
    return destination
```

I believe the correct fix is this:

```python
def assign_by_map(destination: Any, source: Any, attr_map: Dict[str, str]) -> Any:
    for dest_key, source_key in attr_map.items():
        value = dot_getattr(source, source_key, None)
        if value:
            setattr(destination, dest_key, value)
    return destination
```

example attr_map

```python

    attr_map = {
        "terms_of_sale": "terms_of_sale",
        "billing": "freight_billing",
        "com_payment_terms": "commercial_payment_terms",
        "seller_and_buyer_are_related": "seller_and_buyer_are_related",
        "warehouse": "warehouse",
    }
```

### Sourcery Version

v0.7.9

### Code editor or IDE name and version

vscode 1.48.0-insider

### OS name and version

Ubuntu Linux 18.04.4

**Comment by Hellebore** [comment]
Definite bug - thanks for raising a report.

**Comment by Hellebore** [comment]
Fixed with release of 0.7.11
